sb_vision/distance_finding.py

```diff
@@ -0,0 +1,105 @@
+import functools
+from pathlib import Path
+from typing import Tuple, Dict
+
+import cv2
+from lxml import etree
+import re
+import numpy as np
+
+
+def _get_text(element: etree.Element):
+    """parse an xml tag with space-separated variables"""
+    text = []
+    for e in element.itertext():
+        if e.strip():
+            text += re.split('\s+', re.sub('\n', '', e.strip()))
+    return text
+
+
+def get_calibration(file_name: Path) -> Dict:
+    """
+    Parse a calibration xml generated by the camera calibration tool
+    (see https://docs.opencv.org/3.4.0/d7/d21/tutorial_interactive_calibration.html)
+    :param file_name: name of xml file to parse
+    :return: calibrations from the file in a dictionary
+    """
+    calibrations = {}
+    with file_name.open() as file:
+        tree = etree.parse(file)
+        root = tree.getroot()
+        for element in root:
+            if 'type_id' in element.attrib and element.attrib[
+                 'type_id'] == 'opencv-matrix':
+                rows, cols = int(element.find('rows').text), int(
+                    element.find('cols').text)
+                data_type = element.find('dt').text
+                values = _get_text(element.find('data'))
+                if data_type == 'd':  # doubles
+                    data = np.reshape(np.array([float(v) for v in values]),
+                                      (rows, cols))
```

Comment:
What's the benefit of going into numpy here? If there's precision that we're aiming for (numpy's `float64` is more precise than `float` is IIRC), then we should be using `float64` to parse the string. Otherwise IMO we should stick to Python types here (not least as it makes the type signatures much easer to reason about).